src/components/HdToggle.vue

```diff
@@ -50,69 +50,94 @@
   </div>
 </template>
 
-<script>
+<script lang="ts">
+import Vue, { VueConstructor, PropType } from 'vue';
 import OnResizeService from 'homeday-blocks/src/services/on-resize';
 import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
 import HdButton from 'homeday-blocks/src/components/buttons/HdButton.vue';
-import { chevron as chevronIcon } from 'homeday-assets';
-import { kebabMenu as kebabMenuIcon } from 'homeday-assets';
+import { chevron as chevronIcon, kebabMenu as kebabMenuIcon } from 'homeday-assets';
 
 export const TABINDEX_BACKUP_ATTRIBUTE = 'data-hd-tabindex-backup';
 
-export default {
+export interface HdToggleAction {
+  name: string;
+  label: string;
+  icon: string;
+}
+
+export interface HdToggleProps {
+  title: string;
+  canBeToggled: boolean;
+  open: boolean;
+  transitionDuration: number;
+  actions: HdToggleAction[];
+}
```

Comment:
If they are all optional you may want to change like below:
```suggestion
export interface HdToggleProps {
  title?: string;
  canBeToggled?: boolean;
  open?: boolean;
  transitionDuration?: number;
  actions?: HdToggleAction[];
}
```